src/app/parcels/add/[clientId]/page.tsx

```diff
@@ -1,41 +1,101 @@
-import { Metadata } from "next";
-import React from "react";
+"use client";
+
+import React, { useEffect, useState } from "react";
 import ParcelForm, {
     initialParcelFields,
     initialParcelFormErrors,
 } from "@/app/parcels/form/ParcelForm";
-import { getSupabaseServerComponentClient } from "@/supabaseServer";
-import { fetchPackingSlotsInfo, getCollectionCentresInfo } from "@/common/fetch";
+import {
+    CollectionCentresLabelsAndValues,
+    FetchCollectionCentresError,
+    PackingSlotsError,
+    PackingSlotsLabelsAndValues,
+    fetchPackingSlotsInfo,
+    getCollectionCentresInfo,
+} from "@/common/fetch";
+import { ErrorSecondaryText } from "@/app/errorStylingandMessages";
+import { getLocalSupabaseClient } from "../../../../../supabase/tools/getLocalSupabaseClient";
+import Title from "@/components/Title/Title";
 
 interface AddParcelParameters {
     params: {
         clientId: string;
     };
 }
 
-const AddParcels = async ({ params }: AddParcelParameters): Promise<React.ReactElement> => {
-    const supabase = getSupabaseServerComponentClient();
-    const [deliveryPrimaryKey, collectionCentresLabelsAndValues] =
-        await getCollectionCentresInfo(supabase);
-    const packingSlotsLabelsAndValues = await fetchPackingSlotsInfo(supabase);
+const getErrorMessage = (error: FetchCollectionCentresError | PackingSlotsError): string => {
+    let errorMessage = "";
+    switch (error.type) {
+        case "collectionCentresFetchFailed":
+            errorMessage = "Failed to fetch collection centres data.";
+            break;
+        case "packingSlotsFetchFailed":
+            errorMessage = "Failed to fetch packing slots data.";
+            break;
+    }
+    return `${errorMessage} Log ID: ${error.logId}`;
+};
+
+const AddParcels = ({ params }: AddParcelParameters): React.ReactElement => {
+    const [deliveryPrimaryKey, setDeliveryPrimaryKey] = useState<string | null>(null);
+    const [collectionCentresLabelsAndValues, setCollectionCentresLabelsAndValues] =
+        useState<CollectionCentresLabelsAndValues | null>(null);
+    const [packingSlotsLabelsAndValues, setPackingSlotsLabelsAndValues] =
+        useState<PackingSlotsLabelsAndValues | null>(null);
+    const [error, setError] = useState<FetchCollectionCentresError | PackingSlotsError | null>(
+        null
+    );
+    const [isLoading, setIsLoading] = useState(true);
+
+    useEffect(() => {
+        (async () => {
+            setIsLoading(true);
+            const supabase = getLocalSupabaseClient();
```

Comment:
We should be using the "client supabase client", just like the other ones you've already written.